order/main.go

```diff
@@ -119,7 +151,20 @@ func addItemToOrder(c *fiber.Ctx) error {
 		if result2.RowsAffected == 0 {
 			return c.SendStatus(400)
 		} else {
-			return c.SendStatus(200)
+			request_id := shortuuid.New()
+			responses[request_id] = Response{}
+			var payload string = itemId + "/1/" + request_id
+			utils.Publish(payload, mqttClient, "topic/subtractStock")
+
+			//wait for response
+			start := time.Now()
+			for !responses[request_id].has_arrived {
+				if time.Since(start) >= WAIT_TIME {
+					return c.SendStatus(408)
+				}
+				time.Sleep(1 * time.Nanosecond) // to avoid "concurrent map read and map write" error
```

Comment:
Could this potentially still give an error when the read/write takes longer than 1 nanosecond?

Reply:
I am not sure about this or how to test it. but I have tried to implement map as a sync.Map which would be a solution but I was not able to make it work.